subsys/net/ip/tcp.c

```diff
@@ -352,6 +348,32 @@ static void tcp_send(struct net_pkt *pkt)
 	tcp_pkt_unref(pkt);
 }
 
+static int tcp_derive_rto(struct tcp *conn)
+{
+	int rto;
+
+#ifdef CONFIG_NET_TCP_RANDOMIZED_RTO
+	uint32_t gain;
+
+	gain = (uint32_t)conn->rto_random;
+	gain += 1 << 9;
+
+	rto = (gain * CONFIG_NET_TCP_INIT_RETRANSMISSION_TIMEOUT) >> 9;
```

Comment:
I wonder, wouldn't it be easier if the initial retransmission timeout was calculated on the randomization phase (when `sys_rand_get()` is called), so that it doesn't need to be recalculated on each retransmission?

Reply:
We could opt for that option, then you would need at least 2 bytes instead of 1 in the conn struct, what do you prefer?

Reply:
IMO it'd be a bit easier to understand if we stored the timeout instead of some random number used in further computation. And we only need to do the calculation once, when the timeout is re-randomized instead of on each retransmission.

Reply:
Fixed, required a little restructuring, but it becomes much more understandable